pkg/operator/sharded_secret.go

```diff
@@ -32,123 +34,138 @@ import (
 // metadata and the rest of the secret k8s object.
 const MaxSecretDataSizeBytes = v1.MaxSecretSize - 50_000
 
-// ShardedSecret is k8s secret data that is sharded across multiple enumerated
-// k8s secrets. This is used to circumvent the size limitation of k8s secrets.
+// ShardedSecret can shard Secret data across multiple k8s Secrets.
+// This is used to circumvent the size limitation of k8s Secrets.
 type ShardedSecret struct {
-	namePrefix   string
 	template     *v1.Secret
 	data         map[string][]byte
 	secretShards []*v1.Secret
 }
 
-// NewShardedSecret takes a v1.Secret as template and a secret name prefix and
-// returns a new ShardedSecret.
-func NewShardedSecret(template *v1.Secret, namePrefix string) *ShardedSecret {
+// NewShardedSecret takes a v1.Secret object as template and returns a new ShardedSecret.
+// The template's name will be used as the prefix for the concrete secrets.
+func NewShardedSecret(template *v1.Secret) *ShardedSecret {
 	return &ShardedSecret{
-		namePrefix: namePrefix,
-		template:   template,
-		data:       make(map[string][]byte),
+		template: template,
+		data:     make(map[string][]byte),
 	}
 }
 
-// AppendData adds data to the secrets data portion. Already existing keys get
-// overwritten.
-func (s *ShardedSecret) AppendData(key string, data []byte) {
-	if s == nil {
-		return
-	}
+type Byter interface {
+	Bytes() []byte
+}
+
+// Append adds a new key + data pair.
+// If the key already exists, data gets overwritten.
+func (s *ShardedSecret) Append(k fmt.Stringer, v Byter) {
+	s.appendKeyValue(k.String(), v.Bytes())
+}
+
+func (s *ShardedSecret) appendKeyValue(key string, data []byte) {
 	s.data[key] = data
 }
```

Comment:
looks like a shallow function 🤔 

Could we rewrite the function above to:
```go
// Append adds a new key + data pair.
// If the key already exists, data gets overwritten.
func (s *ShardedSecret) Append(k fmt.Stringer, v Byter) {
	s.data[k.String()] = v.Bytes()
}
```
?

Reply:
You're correct. My rationale was that the tests use `appendKeyValue()` so it had to exist anyway but I see your point. I'll fix it.